### PR TITLE
Benchmark docker

### PR DESCRIPTION
**Pull request by brian-intel**
_The PR review is to check for sustainability and correctness.  Sustainability is actually more business critical as correctness is largely tested into the code over time.   Its useful to keep in mind that SW often outlives the HW it was written for and engineers move from job to job so it is critical that code developed for Intel be supportable across many years.  It is up to the submitter and reviewer to look at the code from a perspective of what if we have to debug this 3 years from now after the author is no longer available and defect databases have been lost.  Yes, that happens all the time when we are working with time scales of more than 2 years.  When reviewing your code it is important to look at it from this perspective._

Author Mandatory (to be filled by PR Author/Submitter)
------------------------------------------------------
- Developer who submits the Pull Request for merge is required to mark the checklist below as applicable for the PR changes submitted.  
- Those checklist items which are not marked are considered as not applicable for the PR change.  
- Items marked with an asterisk suffix are mandatory items to check and if not marked will be treated as non-compliant pull requests by the developers for Inner Source Development Model (ISDM) compliance

### PULL DESCRIPTION
FEAT: Moving benchmark tools to Docker containers to all for consistent benchmark environments

### REFERENCES
Reference URL for issue tracking (Github): **\<URL to be filled>**

- [x] Added label to the Pull Request for easier discoverability and search

### CODE MAINTAINABILITY
- [x] **_Commit Message meets guidelines as indicated in the URL\*_**
	- https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md
- [x] **_Every commit is a single defect fix and does not mix feature addition or changes\*_**
- [] Added required new tests relevant to the changes
	- [ ] PR contains URL links to functional tests executed with the new tests 
- [x] Updated Documentation as relevant to the changes
- [x] Updated Build steps/commands changes as relevant
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide description)
- [ ] Specific instructions or information for code reviewers (If any):

Maintainer Mandatory (to be filled by PR Reviewer/Approving Maintainer)
-----------------------------------------------------------------------
- Maintainer who approve the Pull Request for merge are required to mark the checklist below as appropriate for the PR change reviewed as key proof of attestation indicating reasons for merge. 
- Those checklist items which are not marked are considered as not applicable for the PR change. 
- Items marked with an asterisk suffix are mandatory items to check.

### QUALITY CHECKS
- [x] Architectural and Design Fit
- [x] **_Quality of code (At least one should be checked as applicable)\*_**
	- [x] Commit Message meets guidelines
	- [x] PR changes adhere to industry practices and standards
	- [x] Adopted domain specific coding standards ([SBL Coding Standards](https://github.com/tianocore-docs/Docs/raw/master/Specifications/CCS_2_1_Draft.pdf))
	- [x] Error and exception code paths implemented correctly
	- [ ] Code reviewed for domain or language specific anti-patterns
	- [x] Code is adequately commented
	- [x] Code copyright is correct
	- [x] Tracing output are minimized and logic
	- [x] Confusing logic is explained in comments
	- [x] Commit comment can be used to design a new test case for the changes
- [ ] **_Test coverage shows adequate coverage with required CI functional tests pass on all supported platforms\*_**
- [ ] **_Static code scan report shows zero critical issues\*_**

### CODE REVIEW IMPACT
- Summary of Defects Detected in Code Review: **\<%P1*xx,P2*xx,P3*xx,P4*xx%>** \
Note P1/P2/P3/P4 denotes severity of defects found (Showstopper/High/Medium/Low) and xx denotes number of defects found

# _Code must act as a teacher for future developers_

### CODE REVIEW

File: benchmark-scripts/Dockerfile.benchmark [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+RUN echo "Installing PCM" \
+    [ ! -d "$PCM_DIRECTORY" ] && mkdir -p "$PCM_DIRECTORY"
+RUN cd $PCM_DIRECTORY && \
+    git clone --recursive https://github.com/opcm/pcm.git && \
+    ls ${PCM_DIRECTORY} && \
+    cd $PCM_DIRECTORY/pcm  && \
+    mkdir build && \
+    cd build && \
+    cmake .. && \
+    cmake --build .
```
Comment (jim-wang-intel) [inline benchmark-scripts/Dockerfile.benchmark:40]: after successfully installed PCM from git, should we clean it up?

File: benchmark-scripts/Dockerfile.xpu [hunk 2]
```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2021-2022 Intel Corporation.
+# SPDX-License-Identifier: MIT
```
Comment (jim-wang-intel) [inline benchmark-scripts/Dockerfile.xpu:2]: should we use the same copyright and license header as the previous one?

File: benchmark-scripts/Dockerfile.xpu [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+RUN find . -name "pip*" -exec rm -rf {} \; ;exit 0
+RUN find . -name "*normalizer*" -exec rm -rf {} \; ;exit 0
+RUN find . -name "activate*" -exec rm -rf {} \; ;exit 0
+RUN find . -name "Activate*" -exec rm -rf {} \; ;exit 0
+RUN find . -name "python-wheels" -exec rm -rf {} \; ;exit 0
+RUN find . -name "easy_install*" -exec rm -rf {} \; ;exit 0
+RUN find . -name "setuptools*" -exec rm -rf {} \; ;exit 0
+RUN find . -name "__pycache__" -exec rm -rf {} \; ;exit 0
+
+FROM ubuntu:22.04
```
Comment (jim-wang-intel) [inline benchmark-scripts/Dockerfile.xpu:44]: do we need this image layer? it is the same as the base image

File: benchmark-scripts/cleanup_gpu_metrics.sh [hunk 4]
```diff
@@ -39,16 +39,16 @@ fi
...
-    xpumcli dump --rawdata --stop $task_id
-    sudo cp $(jq --raw-output '.dump_file_path' $xpum_file) ${LOG_DIRECTORY}/xpum${device}.csv
-    #sudo cp $(jq --raw-output '.dump_file_path' $xpum_file) j_xpum${device}.csv
-    sudo rm ${LOG_DIRECTORY}/xpum${device}.json
-    cat ${LOG_DIRECTORY}/xpum${device}.csv | \
+    # task_id=$(jq '.task_id' $xpum_file)
+    # xpumcli dump --rawdata --stop $task_id
+    #  cp $(jq --raw-output '.dump_file_path' $xpum_file) ${LOG_DIRECTORY}/xpum${device}.csv
+    # # cp $(jq --raw-output '.dump_file_path' $xpum_file) j_xpum${device}.csv
+    #  rm ${LOG_DIRECTORY}/xpum${device}.json
```
Comment (jim-wang-intel) [inline benchmark-scripts/cleanup_gpu_metrics.sh:49]: remove dead code

File: camera-simulator/camera-simulator.sh [hunk 5]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash -e
 #
-# Copyright (C) 2023 Intel Corporation.
+# Copyright (C) 2022 Intel Corporation.
```
Comment (jim-wang-intel) [inline camera-simulator/camera-simulator.sh:3]: 2023

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]
some minor changes

File: benchmark-scripts/Dockerfile.benchmark (hunk 1 above)
Comment (brian-intel) [reply benchmark-scripts/Dockerfile.benchmark:40]: That's a good idea. Let me add that step

File: benchmark-scripts/Dockerfile.xpu (hunk 2 above)
Comment (brian-intel) [reply benchmark-scripts/Dockerfile.xpu:2]: Yes, let me update to bsd 3

File: benchmark-scripts/cleanup_gpu_metrics.sh (hunk 4 above)
Comment (brian-intel) [reply benchmark-scripts/cleanup_gpu_metrics.sh:49]: done

File: camera-simulator/camera-simulator.sh (hunk 5 above)
Comment (brian-intel) [reply camera-simulator/camera-simulator.sh]: done

File: benchmark-scripts/collect_platform_metrics.sh [hunk 6]
```diff
@@ -35,112 +35,96 @@ then
...
+    docker exec xpum$device bash -c "xpumcli dump --rawdata --start -d 0 -m $metrics -j"
   # DGPU pipeline and  Arc GPU Metrics
   elif [ "$PLATFORM" == "dgpu" ] && [ $HAS_ARC == 1 ]
   then
     echo "==== Starting igt arc ===="
     # Arc is always on Core platform and although its GPU.1, the IGT device is actually 0
     # Collecting both 
-    timeout $DURATION ../docker-run-igt.sh 0
-    timeout $DURATION ../docker-run-igt.sh 1
+    timeout $DURATION docker run -v $SOURCE_DIR/$LOG_DIRECTORY:/$LOG_DIRECTORY -e LOG_DIRECTORY=$LOG_DIRECTORY -itd --privileged benchmark:igt bash -c "/usr/local/bin/intel_gpu_top -d pci:card=0 -J > $LOG_DIRECTORY/igt0.json"
```
Comment (jstein13) [inline benchmark-scripts/collect_platform_metrics.sh:82]: Does it make sense to still have a different container for igt? should we just include it in this one? Maybe in the future.

**Review by jstein13 — APPROVED** [review]

File: benchmark-scripts/collect_platform_metrics.sh (hunk 6 above)
Comment (brian-intel) [reply benchmark-scripts/collect_platform_metrics.sh:79]: For now I think it makes sense to keep them separate but in the future we can possibly merge if the same GPU tool is used for core and dgpu

File: benchmark-scripts/Dockerfile.benchmark [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+# Cleanup
+RUN mkdir -p "/opt/intel/pcm-bin/bin" && mkdir -p "/opt/intel/pcm-bin/lib" && \
+    cp -r "$PCM_DIRECTORY/pcm/build/bin" "/opt/intel/pcm-bin/" && \
+    cp -r "$PCM_DIRECTORY/pcm/build/lib" "/opt/intel/pcm-bin/" && \
```
Comment (jim-wang-intel) [inline benchmark-scripts/Dockerfile.benchmark:45]: should lib copy to lib/ dir?
```suggestion
RUN mkdir -p "/opt/intel/pcm-bin/bin" && mkdir -p "/opt/intel/pcm-bin/lib" && \
    cp -r "$PCM_DIRECTORY/pcm/build/bin" "/opt/intel/pcm-bin/" && \
    cp -r "$PCM_DIRECTORY/pcm/build/lib" "/opt/intel/pcm-lib/" && \
```

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Pull request merged**